apps/factory/src/server/api/routers/prompt.ts

```diff
@@ -127,6 +127,29 @@ export const promptRouter = createTRPCRouter({
       }
     }),
 
+  updateTemplate: protectedProcedure
+    .input(updateTemplateInput)
+    .mutation(async ({ ctx, input }) => {
+      const userId = ctx.jwt?.id as string;
+      try {
+        if (userId) {
+          await ctx.prisma.promptTemplate.update({
```

Comment:
@Rahul917-Kumar remove these checks, as this should be done in the middleware, since userId is needed for all controller actions.